### PR TITLE
Add live password validation to change password dialog

### PR DESCRIPTION
**Pull request by SimonBrandner**
I've added live password validation to `ChangePassword.js`.

### CODE REVIEW

**Comment by SimonBrandner** [comment]
I realized there was a bit more to be done so I added more things. This is what I've added overall:

+ Check if the new password field and confirm password field have the same values
+ Check if the old password field is empty
+ Validate on `Change password` button click

**Comment by t3chguy** [comment]
Needs DCO Sign-off as per https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.rst#sign-off

**Comment by SimonBrandner** [comment]
I am sorry that I forgot the sign-off.

Signed-off-by: Šimon Brandner simon.bra.ag@gmail.com

**Comment by jryans** [comment]
@SimonBrandner Can you include a video / animated GIF of the new experience? That will make it easier to review.

**Comment by SimonBrandner** [comment]
@jryans Sure, here you go:

Peek 2020-11-25 16-05

File: src/components/views/settings/ChangePassword.js [hunk 1]
```diff
@@ -213,29 +347,39 @@ export default class ChangePassword extends React.Component {
                     <form className={this.props.className} onSubmit={this.onClickChange}>
                         <div className={rowClassName}>
                             <Field
+                                ref={field => this[FIELD_OLD_PASSWORD] = field}
                                 type="password"
                                 label={_t('Current password')}
                                 value={this.state.oldPassword}
                                 onChange={this.onChangeOldPassword}
+                                onValidate={this.onOldPasswordValidate}
+                                onBlur={() => CountlyAnalytics.instance.track("onboarding_registration_email_blur")}
```
Comment (t3chguy) [inline src/components/views/settings/ChangePassword.js:356]: this shouldn't be here, this isn't onboarding and there isn't a matching focus track

File: src/components/views/settings/ChangePassword.js [hunk 2]
```diff
@@ -213,29 +347,39 @@ export default class ChangePassword extends React.Component {
...
                                 type="password"
-                                label={_t('New Password')}
+                                label='New Password'
+                                minScore={PASSWORD_MIN_SCORE}
                                 value={this.state.newPassword}
                                 autoFocus={this.props.autoFocusNewPasswordInput}
                                 onChange={this.onChangeNewPassword}
+                                onValidate={this.onNewPasswordValidate}
                                 autoComplete="new-password"
+                                onBlur={() => CountlyAnalytics.instance.track("onboarding_registration_email_blur")}
```
Comment (t3chguy) [inline src/components/views/settings/ChangePassword.js:370]: this shouldn't be here, this isn't onboarding and there isn't a matching focus track

File: src/components/views/settings/ChangePassword.js [hunk 3]
```diff
@@ -213,29 +347,39 @@ export default class ChangePassword extends React.Component {
...
                         <div className={rowClassName}>
                             <Field
+                                ref={field => this[FIELD_NEW_PASSWORD_CONFIRM] = field}
                                 type="password"
                                 label={_t("Confirm password")}
                                 value={this.state.newPasswordConfirm}
                                 onChange={this.onChangeNewPasswordConfirm}
+                                onValidate={this.onNewPasswordConfirmValidate}
                                 autoComplete="new-password"
+                                onBlur={() => CountlyAnalytics.instance.track("onboarding_registration_email_blur")}
```
Comment (t3chguy) [inline src/components/views/settings/ChangePassword.js:382]: this shouldn't be here, this isn't onboarding and there isn't a matching focus track

**Review by t3chguy — CHANGES_REQUESTED** [review]
Other than the spurious CountlyAnalytics tracks it looks sane. Thanks

**Review by nadonomy — APPROVED** [review]

**Comment by SimonBrandner** [comment]
Sorry about that...

**Review by t3chguy — APPROVED** [review]
If you could just pull develop and re-run i18n that would be great, thanks

**Comment by SimonBrandner** [comment]
@t3chguy Done

**Comment by SimonBrandner** [comment]
Thanks

**Comment by t3chguy** [comment]
No, thank you!

**Pull request merged**